neuraxle/steps/flow.py

```diff
@@ -450,6 +450,27 @@ def _make_all_steps_optional(self):
 
         self._refresh_steps()
 
+    def prune(self) -> HyperparameterSamples:
+        """
+        Prune self so as to keep only the chosen step as set in the choice hyperparameter.
+        """
+        # first, get chosen step name. it looks like this:
```

Comment:
It looks like you have copy-pasted [the code in the issue here](https://github.com/Neuraxio/Neuraxle/issues/492#issuecomment-873069923) as-is without modifications. Does it work? I would be surprised that my code works on the first try. I've written it in a hurry and so quickly. Please do a unit test to test the code. 
```suggestion
        # first, get chosen step name. 
```